sci-mathematics/gmp_ecm/gmp_ecm-7.0.4.recipe

```diff
@@ -0,0 +1,77 @@
+SUMMARY="Elliptic Curve Method for Integer Factorization"
+DESCRIPTION="C implementation of the Elliptic Curve Method for factorizing integers"
+HOMEPAGE="http://ecm.gforge.inria.fr/"
+COPYRIGHT="2001-2016 GMP-ECM Authors"
+LICENSE="
+	GNU GPL v3
+	GNU LGPL v3
+	"
+REVISION="1"
+SOURCE_URI="https://gforge.inria.fr/frs/download.php/file/36224/ecm-$portVersion.tar.gz"
+CHECKSUM_SHA256="0cf7b3eee8462cc6f98b418b47630e1eb6b3f4f8c3fc1fb005b08e2a1811ba43"
+SOURCE_DIR="ecm-$portVersion"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion=1.0.0
+libVersionCompat="$libVersion compat >= 1"
+
+PROVIDES="
+	gmp_ecm$secondaryArchSuffix = $portVersion
+	cmd:ecm$secondaryArchSuffix = $portVersion
+	lib:libecm$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libgomp$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	gmp_ecm${secondaryArchSuffix}_devel = $portVersion
+	devel:libecm$secondaryArchSuffix = $libVersion
+	"
+REQUIRES_devel="
+	haiku${secondaryArchSuffix}_devel
+	gmp_ecm$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libgmp$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:cmp
+	cmd:gcc$secondaryArchSuffix
+	cmd:m4
+	cmd:make
+	"
+
+BUILD()
```

Comment:
To add a debuginfo package we just need to insert a call to [definedebuginfopackage](https://github.com/haikuports/haikuports/wiki/HaikuPorter-BuildRecipes#definedebuginfopackage) after BUILD_PREREQUIRES (and before BUILD:
```diff
 	cmd:make
 	"
 
+defineDebugInfoPackage gmp_ecm$secondaryArchSuffix \
+	$libDir/libecm.so.$libVersion
+
 BUILD()
 {
```

Reply:
Same as for `msieve`

Reply:
OK, no problem, let's forget about the debuginfo package for gmp_ecm too.

BTW, I'm sure Haiku needs more skilled developers like you!